src/snowflake/snowpark/_internal/data_source/jdbc.py

```diff
@@ -0,0 +1,582 @@
+        jdbc_ingestion_name = random_name_for_temp_object(TempObjectType.FUNCTION)
+        udtf_table_return_type = ", ".join(
+            [f"{field.name} VARCHAR" for field in self.schema.fields]
+        )
+
+        jdbc_udtf_registration = f"""
+            CREATE OR REPLACE TEMPORARY FUNCTION {jdbc_ingestion_name}(query VARCHAR)
+            RETURNS TABLE ({udtf_table_return_type})
+            LANGUAGE JAVA
+            RUNTIME_VERSION = '{self.java_version}'
+            EXTERNAL_ACCESS_INTEGRATIONS=({self.external_access_integration})
+            {self.imports_sql}
+            {self.packages_sql}
+            {self.secret_sql}
+            HANDLER = 'DataLoader'
+            as
+            $$
+
+            import java.sql.*;
+            import java.util.stream.Stream;
+            import java.util.ArrayList;
+            import java.util.List;
+            import java.util.Map;
+            import java.util.LinkedHashMap;
+            import com.snowflake.snowpark_java.types.*;
+            import com.snowflake.snowpark_java.types.SnowflakeSecrets;
+            import com.snowflake.snowpark_java.types.UsernamePassword;
+
+
+            class OutputRow {{
+                {self.create_output_row_class()}
+            }}
+
+            public class DataLoader{{
+            private final Connection conn;
+
+            private static Connection createConnection() {{
+                try {{
+                    {self.generate_create_connection()}
+                }} catch (Exception e) {{
+                    throw new RuntimeException("Failed to create JDBC connection: " + e.getMessage());
+                }}
+            }}
+
+            public static Class<?> getOutputClass() {{
+                return OutputRow.class;
+            }}
+
+            public DataLoader() {{
+                this.conn = createConnection();
+            }}
+
+            public Stream<OutputRow> process(String query) {{
+                List<OutputRow> list = new ArrayList<>();
+                 try {{
+                    Statement stmt = this.conn.createStatement();
+                    stmt.setQueryTimeout({str(self.query_timeout)});
+                    stmt.setFetchSize({str(self.fetch_size)});
+                    {self.generate_session_init_statement()}
+                    try (ResultSet rs = stmt.executeQuery(query)) {{
+                        ResultSetMetaData meta = rs.getMetaData();
+                        int columnCount = meta.getColumnCount();
+                        while (rs.next()) {{
+                            OutputRow row = new OutputRow();
+                            {self.create_output_row_java_code()}
+                            list.add(row);
+                        }}
+                    }}
+                }} catch (SQLException e) {{
+                    throw new RuntimeException("SQL error: " + e.getMessage(), e);
+                }}
+
+                return list.stream();
```

Comment:
IIUC, this loads all data into local first and then return the whole list, which is likely to cause OOM if the data is large.

I think we should use Stream.generate like: https://medium.com/@orellabac/ingest-external-data-into-snowflake-with-snowpark-and-jdbc-eb487b61078c